core/src/autogluon/core/hpo/ray_hpo.py

```diff
@@ -0,0 +1,457 @@
+import logging
+import os
+import psutil
+import shutil
+
+from ..utils.try_import import try_import_ray
+try_import_ray()  # try import ray before importing the remaining contents so we can give proper error messages
+import ray
+
+from abc import ABC, abstractmethod
+from typing import Optional, Callable, Union, List
+
+from .resources_calculator import ResourceCalculatorFactory
+from .scheduler_factory import SchedulerFactory
+from .searcher_factory import SearcherFactory
+from .. import Space
+
+from ray import tune
+from ray_lightning import RayPlugin
+from ray.tune import PlacementGroupFactory
+from ray.tune.sample import Domain
+from ray.tune.schedulers import TrialScheduler, FIFOScheduler, AsyncHyperBandScheduler, PopulationBasedTraining
+from ray.tune.schedulers.pb2 import PB2
+from ray.tune.suggest import SearchAlgorithm, Searcher
+from ray.tune.suggest.basic_variant import BasicVariantGenerator
+from ray.tune.suggest.hyperopt import HyperOptSearch
+from ray_lightning.tune import get_tune_resources
+
+
+logger = logging.getLogger(__name__)
+
+MIN = 'min'
+MAX = 'max'
+
+searcher_presets = SearcherFactory.searcher_presets
+scheduler_presets = SchedulerFactory.scheduler_presets
```

Comment:
I'm not putting HPO in a class because there's no state that needs to be maintained. In ray tune, the `run` function is also not wrapped in a class. Maybe we can have a `constant.py` file to store these constants.